### PR TITLE
Add sm100 support to <nv/target> for NVCC

### PR DESCRIPTION
**Pull request by wmaxey**
## Description

closes: #1738 

## Checklist

- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by wmaxey** [comment]
pre-commit.ci autofix

File: libcudacxx/include/nv/detail/__target_macros [hunk 1]
```diff
@@ -358,6 +374,14 @@
...
 #  endif
 
+#  undef NV_HAS_FEATURE_SM_100a
+#  define NV_HAS_FEATURE_SM_100a __NV_HAS_FEATURE_SM_100a
+#  if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1000) && defined(__CUDA_ARCH_FEAT_SM100_ALL))
+#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_100a 1
+#  else
+#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_100a 0
+#  endif
+
```
Comment (miscco) [inline libcudacxx/include/nv/detail/__target_macros:384]: I dont think that this is needed yet, as there is no SM100a feature

Comment (ahendriksen) [reply libcudacxx/include/nv/detail/__target_macros:384]: I do think this is needed. SM100a features can build on top of this.

**Review by ahendriksen — APPROVED** [review]
LGTM. 

Question: 
- do we want to have a test to check that it works?
- As a follow-up should we add sm100 to the build matrix?

File: libcudacxx/include/nv/detail/__target_macros (hunk 1 above)
Comment (wmaxey) [reply libcudacxx/include/nv/detail/__target_macros:384]: I think we possibly don't care. If sm100a doesn't exist then it doesn't matter. The only people that would be bothered are those relying on intellisense.

**Comment by wmaxey** [comment]
> LGTM.
> 
> Question:
> 
> * do we want to have a test to check that it works?
> * As a follow-up should we add sm100 to the build matrix?

There is something of a canary existing in the current tests. Compiling with an unknown architecture fails because we try to `CUDA_ARCH` to synthesize the if target query.

**Review by miscco — APPROVED** [review]

**Pull request merged**